doc/tutorials/add_solver.rst

```diff
@@ -0,0 +1,252 @@
+.. _add_solver:
+
+Add a solver to an existing benchmark
+=====================================
+
+This tutorial shows how to add a new solver to a benchmark.
+We illustrate the process on the `Ridge regression benchmark <https://github.com/benchopt/benchmark_ridge>`_ by implementing a solver based on the ``scikit-learn`` Ridge estimator.
+
+.. Hint::
+    If not yet done, you can review the :ref:`get started page <get_started>` to learn how to install benchopt and download an existing benchmark.
+
+
+Before the implementation
+-------------------------
+
+First, create a ``mysolver.py`` file in the ``solvers/`` directory and put inside it the following content:
+
+.. code-block:: python
+    :caption: benchmark_ridge/solvers/mysolver.py
+
+    from benchopt import BaseSolver
+
+    class Solver(BaseSolver):
+        name = 'mysolver'
+
+As you can see, a solver is a Python class, ``Solver``, that inherits from ``benchopt.BaseSolver`` and is declared in a standalone Python file in the benchmark's ``solvers/`` folder.
+The attribute ``name`` does not have to match the file name, but it makes it easier to locate the solver.
+
+Doing the latter steps, our benchmark folder will resemble
+
+.. code-block:: bash
+
+    benchmark_ridge/
+    ├── objective.py     # existing implementation of the Objective
+    ├── datasets/
+    │   ├── ...          # existing datasets
+    └── solvers/
+        ├── mysolver.py  # our newly added solver
+        ├── ...          # other solvers
+
```

Comment:
last comment : I'm not sure we need this anymore, since all snippets have the full path of the file as title